### PR TITLE
[wip] Add information page author visibility option

### LINKED ISSUE #642: Author display
Authors of information pages should only be visible for logged in users.
Displaying the author should also be made optional when editing the document.

### PR DESCRIPTION
**Pull request by LoadingByte**
Authors of information pages are now only visible for logged in users. Displaying the author can also be made optional when editing the document.

This closes #642.

### CODE REVIEW

File: _1327/information_pages/templatetags/information_pages_templatetags.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from django.template import Library
+
+from _1327.information_pages.models import InformationDocument
+
+register = Library()
+
+
+@register.assignment_tag(takes_context=True)
```
Comment (janno42) [inline _1327/information_pages/templatetags/information_pages_templatetags.py]: please register a filter that accesses a new model method `can_user_see_author` instead. have a look at https://github.com/fsr-itse/EvaP/blob/e125a887143848408a5b9c9be77fea53df6d8c77/evap/evaluation/templatetags/evaluation_filters.py#L81 for an example.

you can then use `{% if document.last_author and document|can_user_see_author:request.user %}` in the template.

File: _1327/information_pages/tests.py [hunk 2]
```diff
@@ -222,6 +242,36 @@ def test_last_author(self):
...
+
+		self.document.show_author_to = InformationDocument.SHOW_AUTHOR_TO_EVERYONE
+		with transaction.atomic(), revisions.create_revision():
+			self.document.save()
+			revisions.set_user(self.user)
+			revisions.set_comment('test version 2')
+		self.assertIn(author_str, self.app.get(self.document.get_view_url(), user=self.user).body.decode('utf-8'))
+		self.assertIn(author_str, self.app.get(self.document.get_view_url(), user=self.unauthenticated_user).body.decode('utf-8'))
+
+		# The following part of the test is currently disables as the anonymous user's is_authenticated() method evaluates to true
```
Comment (janno42) [inline _1327/information_pages/tests.py]: @Bartzi any ideas about the tests?

Comment (Bartzi) [reply _1327/information_pages/tests.py]: you could do `context['request'].user.is_authenticated() and not context['request'].user.is_anonymous()` in [this](https://github.com/fsr-itse/1327/pull/649/files/fa16cf141f434cb38f37685c78741e1b0dc7aa70#diff-1b2e97b8158c25ff8530da9c362dc909R13) line

Comment (Bartzi) [reply _1327/information_pages/tests.py]: maybe that helps

**Comment by janno42** [comment]
replaced by #698

**Pull request closed without merging**